### ISSUE TITLE
It keeps loading after querying

### ISSUE THREAD

**Issue by LCehoennardo** [h-sphere/sql-seal#1]
It keeps loading after querying
<img width="859" alt="image" src="https://github.com/h-sphere/sql-seal/assets/41560874/c9c77363-865e-418a-a39e-0ca84f66a170">

**Comment by tlwt** [comment]
Same problem here.  Plugin version 0.2.0 on Mac OS.

![Bildschirmfoto 2024-08-05 um 16 38 27](https://github.com/user-attachments/assets/b1a05604-0d49-40c9-a9b7-8ea19e76a018)

**Comment by tlwt** [comment]
I guess it is realted to the better_sqlite3.node not being downloaded correctly. I downloaded the version for arm manually, allowed it in the security settings, but obsidian does not load afterwards anymore.

**Comment by kulak-at** [comment]
Thank you for the comments. You are right, the issue is with better_sqlite3.node not being available in the plugin folder. I will try to release fix in upcoming days.

**Comment by kulak-at** [comment]
@LCehoennardo , @tlwt : This should be now fixed in the version 0.3.0. Please let me know if that works and open new issues if you find any more problems. It's an early stage of the library so all feedback is really appreciated!

**Comment by tlwt** [comment]
@kulak-at, thanks for the fix. The file is loading now, but Apple M1 is arm architecture and the better_sqlite3.node is build for x86. I briefly checked https://github.com/WiseLibs/better-sqlite3/releases to see if I find the right version and copy it manually, but this does not really seam to work. So stuck at the same "loading" message.

**Comment by kulak-at** [comment]
That's weird - it does work on my Apple Silicon computer with no problem after the fix. Can you check the errors in the console and attach the screenshot here? It might be another problem with is not properly handled in the library.

**Comment by tlwt** [comment]
I just checked and I dont get any error message in the console. I also tried the manual plugin install. The better_node3 loads automatically, but the message is still "loading SQLSeal database".

**Comment by kulak-at** [comment]
Thank you. In that case could you share more details about your setup? Can you share your sqlseal query you are trying to execute? If it’s querying csv, is there a chance you can attach redacted version of this file?
BTW: thank you for your support on KoFi!

Best!

**Comment by tlwt** [comment]
Finally found some error message:

### Setup
- Plugin: SQLSeal 0.3.0 installed via plugin (not manual)
- Obsidian: v1.6.7
- MacBook Pro M1, 16 GB

### MD & CSV

- [SQLSeal.md](https://github.com/user-attachments/files/16597846/SQLSeal.md)
- [test4.csv](https://github.com/user-attachments/files/16597829/test4.csv)

### Error message
I guess key is the "not a match-o".
![Bildschirmfoto 2024-08-13 um 12 34 20](https://github.com/user-attachments/assets/910c2e8c-dbe5-4f8a-b82a-26da1f0e3583)

**Comment by KyattPL** [comment]
I've tried the plugin on the Windows machine with similar result, it seems that packaged better_sqlite3.node file is not supported on Windows:
![image](https://github.com/user-attachments/assets/df275131-6cd9-4d5b-8dca-9a382784472b)

I've tried copying the version manually like @tlwt suggested, with the wrong version the logs show that:
![image](https://github.com/user-attachments/assets/a43b69ae-e09a-41b1-8674-dab4fbb74920)

Then I've tried to download the proper version based on these logs:
[better-sqlite3-v11.2.1-electron-v116-win32-x64.tar.gz](https://github.com/WiseLibs/better-sqlite3/releases/download/v11.2.1/better-sqlite3-v11.2.1-electron-v116-win32-x64.tar.gz)

After the "Loading plugins..." and "Loading vault..." messages, I get a complete black screen:
![image](https://github.com/user-attachments/assets/fc1db1be-26be-48ea-ac6a-a7c055fbaec4)

I can't open the console (unless there's a debug mode for Obsidian) so can't really tell what's going on under the hood.

**Comment by kulak-at** [comment]
Thank you @KyattPL for extra context. I will try to rework the library to work with all architectures, I see where the issue lies now.

**Comment by kulak-at** [comment]
Hey @LCehoennardo @tlwt @KyattPL !
I've just released version 0.4.0 of SQLSeal which should hopefully fix the issue for all of you. The new version downloads appropriate version from the separate repository (which is just a mirror of files from the original project). It should adjust the file based on the architecture and system of your machine. Let me know if it works for you!

Also, a small personal plug: I've just started Discord channel for this project (and other Obsidian plugins I'm working on) - if you have more ideas how we can expand this plugin or just want to chat about Obsidian things, join it! 

https://discord.gg/ZMRnFeAWXb

Best

**Comment by kulak-at** [comment]
Hey, just released version 0.10.0 further addresses this issue. The project has been rewritten to use SQLJS project that uses Web Assembly module rather than native code which should greatly fix compatibility with different systems. Please get back to me if this fixes this issue!

**Comment by kulak-at** [comment]
Closing this as there was no extra activity here since August. If this issue still persists, please open new issue based on the latest version of the library. Thanks!
